### PR TITLE
Introduce prettier to keep the format of the codebase consistent

### PR DESCRIPTION
**Pull request by realityforge**
This PR attempts to address issue #1060.

In short it introduces the `prettier` and `lint-staged` dependencies and setups the same tasks and configuration as in [mobxjs/mobx-state-tree](https://github.com/mobxjs/mobx-state-tree). The prettier was also run across the codebase to ensure it aligns with the new codestyle. 

It should be notes that the codestyle in `mobx-state-tree` is not the same as in this codebase and thus this commit introduces a lot of changes. If you prefer I could align the rules used by prettier to more closely match this repository to minimize the amount of changes in the pull request.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,5 +92,11 @@
     "functional-reactive-programming",
     "state management",
     "data flow"
-  ]
-}
\ No newline at end of file
+  ],
+  "lint-staged": {
+    "*.{ts,tsx,js,jsx}": [
+      "prettier --write --print-width 100 --tab-width 4 --no-semi",
```
Comment (capaj) [inline package.json]: @mweststrate are you okay with no semi and tab with 4?

**Comment by mweststrate** [comment]
Yep! Sorry for not responding earlier

Op ma 7 aug. 2017 10:18 schreef Jiri Spac <notifications@github.com>:

**Comment by capaj** [comment]
awesome. Since this builds fine, I'll merge it.

**Comment by phiresky** [comment]
Am I missing something or is there a dependency to `husky` missing for this to run as a pre-commit script?

**Comment by capaj** [comment]
@phiresky yeah we're missing husky. We have `precommit` script, but that won't run without husky. I'll open a PR.

**Pull request merged**